include/fs/fs_data_types.hpp

```diff
@@ -0,0 +1,51 @@
+#ifndef FS_DATA_TYPES_HPP
+#define FS_DATA_TYPES_HPP
+
+#include "fs_constants.hpp"
+
+size_t write_u64(u_int64_t num, char buf[]);
+
+size_t read_u64(u_int64_t *num, char buf[]);
+
+size_t write_u32(u_int32_t num, char buf[]);

```

Comment:
will we need 32 bit r/w?

Reply:
I made some of the inode metadata 32bit as a proof of concept, it is probibly better to change everything in the final version to match linux semantics.